mps-sync-plugin-lib/src/main/kotlin/org/modelix/mps/sync/SyncServiceImpl.kt

```diff
@@ -243,43 +251,46 @@ class SyncServiceImpl(userNotifier: INotifier) : ISyncService {
     override fun bindModelFromMps(model: SModelBase, branch: IBranch): IBinding {
         logger.info { "Binding Model '${model.name}' to the server." }
 
-        val synchronizer = ModelSynchronizer(branch)
+        val synchronizer = ModelSynchronizer(branch, serviceLocator = serviceLocator)
         // synchronize model. Warning: blocking call
         val binding = synchronizer.addModel(model).getResult().get() as IBinding
         // wait until the model imports are synced. Warning: blocking call
         synchronizer.resolveModelImportsInTask().getResult().get()
 
         if (binding !is EmptyBinding) {
             val message = "Model Binding for '${model.name}' is created."
-            notifierInjector.notifyAndLogInfo(message, logger)
+            notifier.notifyAndLogInfo(message, logger)
         } else {
             val message = "No Model Binding is created for '${model.name}'. This might be due to an error."
-            notifierInjector.notifyAndLogWarning(message, logger)
+            notifier.notifyAndLogWarning(message, logger)
         }
 
         return binding
     }
 
-    override fun setActiveProject(project: Project) {
-        mpsProjectInjector.setActiveProject(project)
-    }
-
-    override fun close() {
-        logger.debug { "Closing SyncServiceImpl." }
-        // dispose task and wait queues
-        SyncQueue.close()
-        FuturesWaitQueue.close()
-        // dispose replicated model
-        BranchRegistry.close()
-        // dispose all bindings
-        BindingsRegistry.deactivateBindings()
-        logger.debug { "SyncServiceImpl is closed." }
-    }
+    private fun registerLanguages(): MPSLanguageRepository {
+        val mpsLanguageRepo = MPSLanguageRepository(mpsRepository)
+        /*
+         * SRepository is not comparable in MPSLanguageRepository, because at runtime it is ProjectRepository that does
+         * not implement equals or hashCode. Therefore, if we run the same MPS multiple times, then the repository will
+         * be registered more than once that confuses Modelix when resolving Concepts.
+         */
+        val comparabeMpsLanguageRepo = object : ILanguageRepository by mpsLanguageRepo {
+            override fun hashCode() = Objects.hashCode(this.getPriority(), this.getAllConcepts().map { it.getUID() })
+            override fun equals(other: Any?): Boolean {
+                return if (other == null || other !is ILanguageRepository) {
+                    false
+                } else if (this.getPriority() != other.getPriority()) {
+                    false
+                } else {
+                    val thisConceptUids = this.getAllConcepts().map { it.getUID() }
```

Comment:
What, if the order of concept differs?

Reply:
deprecated by https://github.com/modelix/modelix.mps-plugins/pull/112#discussion_r1660715076